File: trees_and_graphs/inorder_traversal.py
def inorder_traversal(root):
    stack = [root]
    solution = []
    current = root
    while(stack):
        if current.left:
            stack.append(current.left)
            current = current.left
        else:
            solution.append(current)
            if current.right:
                stack.append(current.right)
    return solution

File: trees_and_graphs/inorder_traversal.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var inorderTraversal = function(root) {
  let stack = [];
  let valueList = [];
  let curr = root;
  while(stack.length || curr) {
    while (curr) {
      stack.push(curr);
      curr = curr.left;
    }
    curr = stack.pop();
    valueList.push(curr.val);
    curr = curr.right;
  }
  return valueList;
};